make/test_target_script.sh

```diff
@@ -0,0 +1,248 @@
+following text file that describes how to run tests for a specific
+application.
+
+EOM
+        echo $ERL_TOP/HOWTO/TESTING.md
+    }
+    print_highlighted_msg_with_printer $YELLOW print_msg
+}
+
+print_all_tests_for_application_notes () {
+    print_msg () {
+        cat << EOM
+
+NOTE 1
+
+ct_run will now attempt to execute tests in the test directory, which
+may take a long time to do. One can pass arguments to ct_run by
+setting the ARGS variable when invoking "make test".
+
+Example:
+
+make ARGS="-suite asn1_SUITE -case ticket_7407" test
+
+NOTE 2
+
+You may want to look at the more established way of running tests that
+is described in the following text file if you encounter strange
+errors:
+
+EOM
+        echo "$ERL_TOP/HOWTO/TESTING.md"
+    }
+    print_highlighted_msg_with_printer $LIGHT_CYAN print_msg
+}
+
+print_c_files_warning () {
+    print_msg () {
+        cat << EOM
+
+WARNING
+
+The test directory contains .c files which means that some test cases
+will probably not work correctly when run through "make test". The
+text file at the following location describes how one can compile and
+run all test cases:
+
+
+EOM
+        echo $ERL_TOP/HOWTO/TESTING.md
+    }
+    print_highlighted_msg_with_printer $YELLOW print_msg
+}
+
+
+print_on_error_note () {
+    print_msg () {
+        cat << EOM
+NOTE:
+
+Some test cases do not work correctly when run through "make test" as
+they are designed to be run through the method that is described in
+the "$ERL_TOP/HOWTO/TESTING.md" text file. You may want to check this
+text file if you encounter strange errors. Note also that you can
+rerun a specific test case by passing parameters to ct_run as in the
+example below:
+
+make ARGS="-suite asn1_SUITE -case ticket_7407" test
+
+EOM
+    }
+    print_highlighted_msg_with_printer $NC print_msg
+}
+
+# Check ERL_TOP
+
+if [ -d "$1" ]
+then
+    ERL_TOP="$1"
+    shift
+fi
+
+if [ -z $ERL_TOP ]
+then
+    ERL_TOP=`git rev-parse --show-toplevel`
+    if [ $? = 0 ]
+    then
+        print_highlighted_msg $LIGHT_CYAN "The environment variable ERL_TOP has been set to the git root"
+    else
+        echo "The ERL_TOP environment variable need to be set before this script is executed."
+        exit 1
+    fi
+fi
+
+export ERL_TOP=$ERL_TOP
+
+
+if [ -z "${ARGS}" ]
+then
+   ARGS="$@"
+fi
+
+# make test in root
+DIR=`pwd`
+if [ "$DIR" -ef "$ERL_TOP" ]
+then
+    TARGET_SYS=`$ERL_TOP/erts/autoconf/config.guess`
+    REL_DIR="$ERL_TOP/release/$TARGET_SYS"
+    cd "$REL_DIR"
+    ./Install -sasl `pwd`
+    export PATH="$REL_DIR/bin:$PATH"
+    cd "$ERL_TOP/release/tests/test_server"
+    print_all_tests_takes_long_time_warning
+    echo "The tests will start in a few seconds..."
+    sleep 45
+    cd "$ERL_TOP/release/tests/test_server"
+    erl -eval "ts:install(),erlang:halt()"
+    erl -noinput -eval "ts:run([all_tests,batch]),erlang:halt()"
+    exit $?
+fi
+
+# make test in application directory
+if [ -d test ]
```

Comment:
Is this test needed?

Reply:
No, but I think it may still be useful if one wants to invoke the script directly without using `make test`.